mpi-proxy-split/mpi-wrappers/generate-mpi-fortran-wrappers.py

```diff
@@ -90,13 +90,17 @@ def emit_wrapper(decl, ret_type, fnc, args, arg_vars):
   if fnc in ['MPI_Wtime', 'MPI_Wtick']:
     print("  return " + fnc + "(" + cargs + ");")
   else:
+      if "int* index" in fargs:
+        # This is a temporary fix for the Fortran-to-C interface bug -
+        # more details can be found in MPI_Testany in mpi_request_wrappers.cpp
+        print("  int *local_index = index;");
       print("  *ierr = " + fnc + "(" + cargs + ");")
       if "int* index" in fargs:
         if not (fnc in ['MPI_Waitany', 'MPI_Testany']):
           warnings.warn("'int* index' found in {fnc} - likely to cause erroneous behavior".format(fnc=fnc))
         # We need this line because Fortran arrays start from 1, and *index
         # refers to the index of the first request that completes in an array.
-        print("  *index = *index + 1;")
+        print("  *local_index = *local_index + 1;")
```

Comment:
I'm confused why it makes a difference.

Unlike `Testall`&`Watiall` below, although you define a local variable `local_index` here, you don't pass the address of this local variable to the real MPI function. You still use the same "memory address" before&after, Why it can help?

Reply:
The function originally looks like this:
```
EXTERN int mpi_testany_(..., int *index, ...) {
  *ierr = MPI_Testany(..., int *index, ...);
  *index = *index + 1;
  return *ierr;
}
```
But the bug affects the addresses passed into the functions, so after another function inside the `MPI_Testany` call returns, all references to `index` get jumbled up. We already have a fix for this within the `MPI_Testany` C wrapper, but because we have to do an operation on `index` outside of the C wrapper (we only increment `index` if Fortran calls it), we need a local copy of `index` that we operate on after the C wrapper returns.